### ISSUE TITLE
Autocomplete Doesn't Announce Available Results on Screen Readers

### ISSUE THREAD

**Issue by bbertucc** [bbertucc/equalify#374]
When I'm using a screen reader, I have no way of knowing how many results are available as I search through filters. I also can't select a filter using the keyboard navigation.

That is different than what we have on managed.equalify.app. On managed, we use the https://github.com/alphagov/accessible-autocomplete. That announces how many results are available as we type and allows us to use keyboard navigation to select results.

I've created this video to show you the difference when using Apple VoiceOver: https://github.com/alphagov/accessible-autocomplete

Adding @kevinandrews1 to confirm this issue, before I turn it over to development.

**Comment by kevinandrews1** [comment]
Confirmed! I'll assign it to @wilsuriel03 but feel free to reassign as needed.

**Comment by bbertucc** [comment]
Thanks. @wilsuriel03 let me know if you need further clarification.

**Comment by wilsuriel03** [comment]
on it! I will have a fix for my remaining Issues soon @bbertucc .

**Comment by bbertucc** [comment]
Amazing @wilsuriel03!

**Comment by wilsuriel03** [comment]
This has been successfully fixed. @bbertucc . Feel free to retest @kevinandrews1

**Comment by bbertucc** [comment]
Looks awesome to me @wilsuriel03! Assigning to @kevinandrews1 to do final testing.

**Comment by kevinandrews1** [comment]
@bbertucc @wilsuriel03 is the intent to be able to selecgt multiple items? For example, when I select messages when editing a report, it seems the intention is to be able to select only one message at a time. The focus behavior is weird--if I close the list, tab off the autocomplete, then return to it, the list of items seems exposed yet you don't actually know what's selected or perhaps the selection is cleared? It's also possible I just have no idea how this component is supposed to function. Here's a video: https://drive.google.com/file/d/1sPjrindAlMIANJxFWwNTkMlpjyPlALVe/view?usp=sharing

**Comment by bbertucc** [comment]
@kevinandrews1, a few questions
- Would it make more sense to you if you see a "Selected Filters" heading in a selected filters landmark region when you get to the page with a list of selected filters that you can browse and remove?
- Should we add description text to the filter autocomplete field that says "Search and browse filters then add multiple filters to reports."? Does that clarify the intent?
- When you select a filter, would the process be cleared up if we announce this when you select a filter: "Filter Name was added to Selected Filters. Type to search for additional filters. Use the arrow keys to select from results. Save your updates by clicking the Update Report button."

Also, @wilsuriel03, looks like the selected filters and their "remove filter" button cannot be navigated to using the tab key. That is a big problem that should be fixed. We'll wait to hear from Kevin on the other issues.

-Bb.

**Comment by kevinandrews1** [comment]
@bbertucc @wilsuriel03 I'll address each point in turn. These are all great suggestions. I was considering acceptable solutions given time constraints. Implementing any but ideally all of these will significantly enhance accessibility and usability. I would stress the inability to remove selected filters with the keyboard is a critical barrier and should be addressed asap prior to v1 even if nothing else is addressed. All elements *must* be keyboard accessible. Also fun fact - I'm not a dev but know enough code to be dangerous. Hopefully this is helpful to you:
## Selected Filters Heading
Having a "Selected Filters" heading within a selected filters landmark region is a great idea. It provides clear semantic structure and context, making it easier for users, especially those using screen readers, to understand the purpose of the section and navigate through it. Take it a step further and make the filters native checkboxes for even easier navigability and straightforward ease of use.
```
<div role="region" aria-labelledby="selected-filters-heading">
  <h2 id="selected-filters-heading">Selected Filters</h2>
  <ul id="selected-filters-list">
    <!-- Dynamically added selected filters -->
  </ul>
</div>
```
## Descriptive Text
Adding descriptive text such as "Search and browse filters then add multiple filters to reports" is beneficial. It provides users with a clear understanding of what the field does and what actions they can perform.
Announcing when a filter is added, for example, "Filter Name was added to Selected Filters. Type to search for additional filters. Use the arrow keys to select from results. Save your updates by pressing the Update Report button," is very helpful. It ensures that users are aware of the changes they make and guides them on the next steps.
```
<label for="filter-autocomplete">Filter Autocomplete</label>
<input type="text" id="filter-autocomplete" aria-describedby="filter-description">
<p id="filter-description">Search and browse filters then add multiple filters to reports.</p>
```
And then the filter selection:
```
<div role="status" aria-live="polite" id="filter-announcement" class="sr-only"></div>
function addFilter(filterName) {
  // Code to add the filter to the selected filters list
  // ...
  // Update the aria-live region to announce the addition
  const announcement = `Filter ${filterName} was added to Selected Filters. Type to search for additional filters. Use the arrow keys to select from results. Save your updates by clicking the Update Report button.`;
  document.getElementById('filter-announcement').textContent = announcement;
}
```
## Keyboard Only
The fact that selected filters and their "remove filter" buttons cannot be navigated using the tab key is a significant accessibility issue that should be addressed promptly. I didn't know about those buttons which is a testament to its severity. Keyboard navigability is crucial for users who rely on keyboard interaction, including those using screen readers or other assistive technologies.
```
<div role="region" aria-labelledby="selected-filters-heading">
  <h2 id="selected-filters-heading">Selected Filters</h2>
  <ul id="selected-filters-list">
    <li>
      <span>Filter 1</span>
      <button type="button" onclick="removeFilter('Filter 1')">Remove</button>
    </li>
    <li>
      <span>Filter 2</span>
      <button type="button" onclick="removeFilter('Filter 2')">Remove</button>
    </li>
    More filters...
  </ul>
</div>
function removeFilter(filterName) {
  // Code to remove the filter from the selected filters list
  // ...
  // Announce the removal (optional, if needed)
  const announcement = `Filter ${filterName} was removed from Selected Filters.`;
  document.getElementById('filter-announcement').textContent = announcement;
}
```

**Comment by bbertucc** [comment]
Thanks @kevinandrews1. There are way too many filters to display them as checkboxes.

Closing this issue in favor of an issue with concise fixes here: https://github.com/EqualifyEverything/equalify/issues/398

**Comment by kevinandrews1** [comment]
Fair @bbertucc. Pagination is an option but probably overengineering. The autocomplete is fine jsut have to make sure everything is announced and kb operable.
